starknet_py/net/full_node_client.py

```diff
@@ -387,22 +431,24 @@ async def get_contract_nonce(
 def get_block_identifier(
     block_hash: Optional[Union[Hash, Tag]] = None,
     block_number: Optional[Union[int, Tag]] = None,
-) -> dict:
+) -> Union[Dict, Tag, int, Hash, None]:
     if block_hash is not None and block_number is not None:
         raise ValueError(
             "Arguments block_hash and block_number are mutually exclusive."
         )
 
     if block_hash in ("latest", "pending") or block_number in ("latest", "pending"):
-        return {"block_id": block_hash or block_number}
+        return block_hash or block_number
 
     if block_hash is not None:
-        return {"block_id": {"block_hash": _to_rpc_felt(block_hash)}}
+        ret = {"block_hash": _to_rpc_felt(block_hash)}
+        return ret
 
     if block_number is not None:
-        return {"block_id": {"block_number": block_number}}
+        ret = {"block_number": block_number}
+        return ret
 
-    return {"block_id": "pending"}
+    return "pending"
```

Comment:
Why was this method changed?

Reply:
The [BLOCK_ID schema](https://github.com/starkware-libs/starknet-specs/blob/ed85de6a701c966ec7395b0df4c8bd62f3cddbcd/api/starknet_api_openrpc.json#L771) is also used in the [getEvents EVENT_FILTER](https://github.com/starkware-libs/starknet-specs/blob/ed85de6a701c966ec7395b0df4c8bd62f3cddbcd/api/starknet_api_openrpc.json#L580) but with a different key.  I [originally added a new parameter](https://github.com/software-mansion/starknet.py/pull/801/commits/2e6e293c6c8625b41d0319d2a4f1c19af36d94c9) to define the key, but found it excessive so modified this method to return only the BLOCK_ID matching object and let the consumer define the key used.

Reply:
Okay, makes sense. I didn't notice that at first.